From 7e5f89199d2b13f3858aedfce00d61c78547e034 Mon Sep 17 00:00:00 2001
From: Hui Peng <phui@google.com>
Date: Sat, 14 Jan 2023 02:54:06 +0000
Subject: [PATCH] Fix an OOB bug in register_notification_rsp

Bug: 245916076
Test: manual
Ignore-AOSP-First: security
Change-Id: I901d973a736678d7f3cc816ddf0cbbcbbd1fe93f
(cherry picked from commit daa3efc5e53c8613f4b1a33e095ff6c6460b8d02)
Merged-In: I901d973a736678d7f3cc816ddf0cbbcbbd1fe93f
---
 system/btif/src/btif_rc.cc | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/system/btif/src/btif_rc.cc b/system/btif/src/btif_rc.cc
index f6d6b036a5..ad05be2063 100644
--- a/system/btif/src/btif_rc.cc
+++ b/system/btif/src/btif_rc.cc
@@ -1959,6 +1959,11 @@ static bt_status_t register_notification_rsp(
                    dump_rc_notification_event_id(event_id));
   std::unique_lock<std::mutex> lock(btif_rc_cb.lock);
 
+  if (event_id > MAX_RC_NOTIFICATIONS) {
+    BTIF_TRACE_ERROR("Invalid event id");
+    return BT_STATUS_PARM_INVALID;
+  }
+
   memset(&(avrc_rsp.reg_notif), 0, sizeof(tAVRC_REG_NOTIF_RSP));
 
   avrc_rsp.reg_notif.event_id = event_id;
-- 
2.39.1.456.gfc5497dd1b-goog

